pin/views.py

```diff
@@ -1,4 +1,316 @@
-from django.shortcuts import render
-from django.views       import View
-from django.http        import JsonResponse
+import boto3
+import json
+import jwt
 
+from django.shortcuts       import render
+from django.views           import View
+from django.http            import JsonResponse, HttpResponse
+from django.forms.models    import model_to_dict
+from django.core            import serializers
+from django.db.models       import Q
+from django.core.exceptions import ObjectDoesNotExist
+
+from pin.models         import (
+    Topic,
+    UserTopic,
+    Board,
+    Pin,
+    Topic,
+    PinUser,
+    PinTopic,
+    Comment
+)
+from account.models         import User, Follow, Social
+from pinterrorist.settings  import S3URL, SECRET_KEY, ACCESS_KEY, SECRET_ACCESS_KEY
+
+def login_required(func):
+        def wrapper(self, request, *args, **kwargs):
+            
+            header_token            = request.headers['Authorization']
+            print(header_token)
+            if header_token         == None:
+                return JsonResponse({"message": "THE TOKEN IS EMPTY"}, status=402)
+            decoded_token           = jwt.decode(header_token, SECRET_KEY, algorithm='HS256')['id']
+            print(decoded_token)
+            user                    = User.objects.get(id=decoded_token)
+            user_id                 = user.id
+            kwargs['user']          = user
+            kwargs['user_id']       = user_id
+            
+            try:
+                if User.objects.filter(id=decoded_token).exists():
+                    return func(self, request, *args, **kwargs)
+                else:
+                    return JsonResponse({"message": "THE USER DOES NOT EXIST"})
+            except jwt.DecodeError:
+                return JsonResponse({"message": "WRONG_TOKEN!"}, status=403)
+            except KeyError:
+                return JsonResponse({"message": "KEY ERROR"}, status=405)
+        return wrapper
+
+class IntroTopicView(View):
+    
+    def get(self, request, *args, **kwargs):
+        topiclst        = list(Topic.objects.values())
+        return JsonResponse({"topics": topiclst})
+
+    @login_required
+    def post(self, request, *args, **kwargs):
+        body            = json.loads(request.body)
+        if body == None:
+            return JsonResponse({"message": "THERE IS NOTHING IN THE REQUEST BODY"}, status=405)
```

Comment:
이것도 바디에 빈거를 준거면 400에러 아닌가요 ? 
```suggestion
            return JsonResponse({"message": "THERE IS NOTHING IN THE REQUEST BODY"}, status=405)
```